stan/math/prim/fun/gaus_interp.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef STAN_MATH_PRIM_FUN_GAUS_INTERP
+#define STAN_MATH_PRIM_FUN_GAUS_INTERP
+
+#include <stan/math/prim/fun/conv_gaus_line.hpp>
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+namespace stan {
+namespace math {
+
+// set tolerance for how close interpolated values need to be to
+// the specified function values
+const double INTERP_TOL = 1e-8;
+const double SIG2_SCALE = 0.1;
+
+/*
+given a set of points (x_i, y_i) with x_i's in increasing order, find
+a_i, b_i such that the line between (x_i, y_i) and (x_{i+1}, y_{i+1}) is
+f(t) = a_i*t + b_i, store the a_i's and b_i's in as and bs.
+*/
+void lin_interp_coefs(int n, std::vector<double> xs, std::vector<double> ys,
+                      std::vector<double>& as, std::vector<double>& bs) {
+  // initialize size of vectors of linear coefficients
+  as.resize(n - 1);
+  bs.resize(n - 1);
+
+  // find slope and intercept between each point
+  for (int i = 0; i < n - 1; i++) {
+    as[i] = (ys[i + 1] - ys[i]) / (xs[i + 1] - xs[i]);
+    bs[i] = -xs[i] * as[i] + ys[i];
+  }
+}
+
+/*
+linear interpolation at one point, x, given the coefficients
+*/
+double lin_interp_pt(int n, vector<double> xs, vector<double> ys,
+                     vector<double> as, vector<double> bs, double x) {
```

Comment:
These should all be const& parameters so copies are not made

Reply:
This is true for everywhere you are using std::vector, arithmetic types can be passed by value

Reply:
I made these changes generally and also I created a new file for lin_interp. It seemed to make sense to separate the linear interpolation from this smooth interpolation.